sources/common/src/TxtParser.cpp

```diff
@@ -32,38 +32,16 @@
 #include "TxtParser.h"
 #include "SignalFactory.h"
 
-const std::string PARSER::TxtParser::LINE_COUNTER_SUFFIX = "-LineCounter";
-
-PARSER::TxtParser::TxtParser(const std::string &filename,
-                             const std::string &timeBase,
-                             SIGNAL::SourceRegistry &sourceRegistry,
-                             const std::string &lineCounter,
+PARSER::TxtParser::TxtParser(const std::string &rFilename,
+                             const std::string &rTimeBase,
+                             SIGNAL::SourceRegistry &rSourceRegistry,
                              bool verboseMode) :
-    LogParser(filename, sourceRegistry, verboseMode),
+    LogParser(rFilename, rSourceRegistry, verboseMode),
     m_ValidLines(0),
     m_InvalidLines(0),
-    m_SourceHandle(sourceRegistry.Register(filename)),
-    m_LineCounter(lineCounter),
-    m_LineCounterEnabled(!lineCounter.empty()),
-    m_LineCounterSourceHandle(SIGNAL::SourceRegistry::BAD_HANDLE)
+    m_SourceHandle(rSourceRegistry.Register(rFilename))
 {
-    m_pSignalDb = std::make_unique<SIGNAL::SignalDb>(timeBase);
-
-    // Check if the line counter has been enabled
-    if (m_LineCounterEnabled)
-    {
-        m_LineCounterSourceHandle = sourceRegistry.Register(m_FileName + LINE_COUNTER_SUFFIX);
-    }
-
-    // Process the log
-    Parse();
-
-    // Dump the line counting information.
-    if (m_LineCounterEnabled)
-    {
-        m_LineCounter.RecordToSignalDb(*(m_pSignalDb.get()),
-                                       m_LineCounterSourceHandle);
-    }
+    m_pSignalDb = std::make_unique<SIGNAL::SignalDb>(rTimeBase);
```

Comment:
It couldn't be done there as the base class members cannot be directly initialized on the init list of the inheriting class. But I liked the idea and moved this initialization to the LogParser constructor's init list.